lms/djangoapps/discussion/views.py

```diff
@@ -300,10 +308,14 @@ def single_thread(request, course_key, discussion_id, thread_id):
     """
     course = get_course_with_access(request.user, 'load', course_key, check_if_enrolled=True)
     request.user.is_community_ta = utils.is_user_community_ta(request.user, course.id)
+
     if request.is_ajax():
         cc_user = cc.User.from_django_user(request.user)
         user_info = cc_user.to_dict()
         is_staff = has_permission(request.user, 'openclose_thread', course.id)
+        if not _is_discussion_visible_to_team_user(request, course, discussion_id):
+            return HttpResponseForbidden(TEAM_PERMISSION_MESSAGE)
+
```

Comment:
Refactoring suggestion:
* Rename `_is_discussion_visible_to_team_user` to `_check_team_discussion_access`
* Raise `DiscussionHiddenFromUserException` inside that function instead of returning a bool
* Then, just call `_check_team_discussion_access(request, course, discussion_id)` and handle as necessary

It may make the code more consistent. If you disagree feel free to keep this as-is :)